src/ILCompiler.ReadyToRun/src/ObjectWriter/R2RPEBuilder.cs

```diff
@@ -301,8 +381,109 @@ public void Write(Stream outputStream)
                 CorHeaderFileOffset,
                 outputStream);
 
+            // On Linux, we must patch the section headers. This is
```

Comment:
Nit: It would be nice to break these two chunks of logic into methods that make it clear what the intent is and keep `Write` readable.

Reply:
Good point, fixed in 6th commit.